### PR TITLE
Delete Legacy HTTPS Docs

### PR DESCRIPTION
**Pull request by rachelwhitton**
CC @ari-gold 
Retire legacy docs and update links throughout

## Required Redirects Post Launch
- [ ] /docs/build-https => /docs/https
- [ ] /docs/domains => /docs/guides/launch/domains
- [ ] /docs/enable-https-vanity-domain => /docs/https
- [ ] /docs/enable-https => /docs/https
- [ ] /docs/ssl-tls => /docs/https
- [ ] /docs/videos/domains => /docs/guides/launch/domains

### CODE REVIEW

File: source/_docs/build-https.md [hunk 1]
```diff
@@ -1,17 +0,0 @@
----
-title: Developing with HTTPS
```
Comment (ari-gold) [inline source/_docs/build-https.md]: I think we need to preserve this content somewhere, but maybe we can deal with that in http-to-https

File: source/docs.html [hunk 2]
```diff
@@ -161,7 +161,7 @@ <h2 class="subtitle">Popular Documentation</h2>
         <li><a href="/docs/terminus/">Terminus Manual</a></li>
         <li><a href="/docs/guides/getting-started/">Getting Started Guide</a></li>
         <li><a href="/docs/platform-considerations/">Platform Considerations</a></li>
-        <li><a href="/docs/domains/">Domains and DNS</a></li>
+        <li><a href="/docs/guides/launch/">Luanch Essentials</a></li>
```
Comment (ari-gold) [inline source/docs.html]: typo

File: source/_docs/build-https.md (hunk 1 above)
Comment (rachelwhitton) [inline source/_docs/build-https.md]: What specifically do we need here that isn't already in http-to-https?

**Comment by ari-gold** [comment]
+1 :shipit: 🚢  🚀

**Pull request merged**